extensions/amp-access-laterpay/0.1/amp-access-laterpay.js

```diff
@@ -28,28 +87,366 @@ export class LaterpayVendor {
     this.accessService_ = accessService;
-    // TODO: implement
+
+    /** @const @private {!Window} */
+    this.win_ = this.accessService_.win;
+
+    /** @const @private {!Document} */
+    this.doc_ = this.win_.document;
+
+    /** @private @const {!Viewport} */
+    this.viewport_ = viewportForDoc(this.win_.document);
+
+    /** @const @private {!LaterpayConfigDef} */
+    this.laterpayConfig_ = this.accessService_.getAdapterConfig();
+
+    /** @const @private {!PurchaseConfigDef} */
+    this.purchaseConfig_ = null;
+
+    /** @const @private {?Function} */
+    this.purchaseButtonListener_ = null;
+
+    /** @const @private {?Function} */
+    this.alreadyPurchasedListener_ = null;
+
+    /** @const @private {!Array<Event>} */
+    this.purchaseOptionListeners_ = [];
+
+    /** @const @private {!boolean} */
+    this.containerEmpty_ = true;
+
+    /** @const @private {?Node} */
+    this.selectedPurchaseOption_ = null;
+
+    /** @const @private {?Node} */
+    this.purchaseButton_ = null;
+
+    const configUrl = (
+      (getMode().localDev || getMode().development) &&
+      this.laterpayConfig_.configUrl
+    ) ? this.laterpayConfig_.configUrl
+      : CONFIG_URL;
+
+    /** @private {string} */
+    this.currentLocale_ = this.laterpayConfig_.locale || 'en';
+
+    /** @private {Object} */
+    this.i18n_ = Object.assign(DEFAULT_MESSAGES,
+                  this.laterpayConfig_.localeMessages || {});
+
+    /** @private {string} */
+    this.purchaseConfigBaseUrl_ = configUrl + CONFIG_BASE_PATH;
+    const articleId = this.laterpayConfig_.articleId;
+    if (articleId) {
+      this.purchaseConfigBaseUrl_ += '&article_id=' + articleId;
+    }
+
+    /** @const @private {!Timer} */
+    this.timer_ = timerFor(this.win_);
+
+    /** @const @private {!Vsync} */
+    this.vsync_ = vsyncFor(this.win_);
+
+    /** @const @private {!Xhr} */
+    this.xhr_ = xhrFor(this.win_);
+
+    installStyles(this.win_.document, CSS, () => {}, false, TAG);
   }
 
   /**
    * @return {!Promise<!JSONType>}
    */
   authorize() {
-    const win = this.accessService_.win;
-    user().assert(isExperimentOn(win, 'amp-access-laterpay'),
+    user().assert(isExperimentOn(this.win_, TAG),
         'Enable "amp-access-laterpay" experiment');
-    // TODO: implement
-    return Promise.resolve({access: true});
+    return this.getPurchaseConfig_()
+    .then(response => {
+      if (response.status === 204) {
+        throw user()
+          .createError('No merchant domains have been matched for this ' +
+            'article, or no paid content configurations are setup.');
+      }
+
+      if (this.laterpayConfig_.scrollTopAfterAuth) {
+        this.vsync_.mutate(() => this.viewport_.setScrollTop(0));
+      }
+      this.emptyContainer_();
+      return {access: response.access};
+    }, err => {
+      const status = err && err.response && err.response.status;
+      if (status === 402) {
+        this.purchaseConfig_ = err.responseJson;
```

Comment:
This is set by the `xhr` service, in the callback used in the `fetchJson` method: https://github.com/ampproject/amphtml/blob/master/src/service/xhr-impl.js#L436

Is that what you meant?